java/fory-core/src/main/java/org/apache/fory/util/ExceptionUtils.java

```diff
@@ -61,6 +61,9 @@ public static RuntimeException handleReadFailed(Fory fory, Throwable t) {
       // carry with read objects for better trouble shooting.
       List<Object> objects = Arrays.asList(readObjects.objects).subList(0, readObjects.size);
       throw new DeserializationException(objects, t);
+    } else if (t instanceof Exception) {
```

Comment:
@chaokunyang I could modify this method to add an extra param that is the class instance that deserializeJavaObject method has access to - this would mean that I could improve the exception message by including the class name. I could overload the existing method by having a new method that takes the extra param and leave this version intact.